code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -704,17 +704,27 @@ func (sc *StorageSmartContract) populateGenerateChallenge(
 	balances c_state.StateContextI,
 ) (*challengeOutput, error) {
 
+	const maxBCPartitionSelect = 5
+
+	var bcItem BlobberChallengeNode
+	var maxUsedCap int64
+
 	var blobberChallenges []BlobberChallengeNode
 	err := blobberChallengeList.GetRandomItems(balances, challRand, &blobberChallenges)
 	if err != nil {
 		return nil, common.NewError("generate_challenges",
 			"error getting random slice from blobber challenge partition")
 	}
 
-	randomIndex := challRand.Intn(len(blobberChallenges))
-	bcItem := blobberChallenges[randomIndex]
-	Logger.Debug("generate_challenges", zap.Int("random index", randomIndex),
-		zap.String("blobber id", bcItem.BlobberID), zap.Int("blobber challenges", len(blobberChallenges)))
+	for i := 0; i < maxBCPartitionSelect; i++ {
+		randomIndex := challRand.Intn(len(blobberChallenges))
+		tempItem := blobberChallenges[randomIndex]
```

Comment:
We don't have to get random item if the total blobber number is less than 5, in that case we can iterate through the blobberChallenges directly.

Reply:
still in partition size of 50, we need to select 5 blobbers randomly, no?

Reply:
Think about the case that partition size of 50, total blobbers is 4, maxBCPartitionSelect is 5, do we need to get the 4 items 5 times randomly? No, we can iterate through the blobbers list directly. This is a corner case which may not happen in really, but we need to know it.

Reply:
So if len(partition) <= maxBCpartitionSelect, we should iterate.. else random select, right?

Reply:
yes